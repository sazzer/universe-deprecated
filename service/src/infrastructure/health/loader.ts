import { ServerLoader, getRouter } from "../server";

import { Healthcheck } from "./check";
import { checkHealth } from "./http";

/** Key into the loader for the individual healthchecks */
const HEALTHCHECKS_KEY = Symbol("health:checks");

/** The loader as seen from the Healthchecks point of view */
export interface HealthLoader {
  [HEALTHCHECKS_KEY]: { [name: string]: Healthcheck };
}

/**
 * Add a new healthcheck to the loader
 * @param loader the loader to register the healthcheck with
 * @param name the name of the healthcheck
 * @param check the actual healthcheck
 */
export function addHealthCheck<T extends HealthLoader>(
  loader: T,
  name: string,
  check: Healthcheck
) {
  if (loader[HEALTHCHECKS_KEY] === undefined) {
    loader[HEALTHCHECKS_KEY] = {};
  }
  loader[HEALTHCHECKS_KEY][name] = check;
}

/**
 * Actually build the healthchecks
 * @param loader the loader to populate
 */
export function build<T extends HealthLoader & ServerLoader>(loader: T) {
  const router = getRouter(loader);

  router.get("/health", checkHealth());
}
