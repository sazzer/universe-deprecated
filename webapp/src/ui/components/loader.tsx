import React from "react";
import { useTranslation } from "react-i18next";

/** The props for the Loader component */
export interface LoaderProps {
  loading: boolean;
}

/**
 * Simple component to indicate that a page is still loading
 */
export const Loader: React.FC<LoaderProps> = ({ loading, children }) => {
  const { t } = useTranslation();

  if (loading) {
    return (
      <div className="d-flex justify-content-center">
        <div className="spinner-border" role="status">
          <span className="sr-only">{t("loader.label")}</span>
        </div>
      </div>
    );
  } else {
    return <>{children}</>;
  }
};
