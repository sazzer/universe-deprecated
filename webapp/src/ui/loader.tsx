import React from "react";
import { useTranslation } from "react-i18next";

/**
 * Loader to display whilst the page is still loading
 */
export const Loader: React.FC = () => {
  const { t } = useTranslation();

  return (
    <div className="d-flex justify-content-center">
      <div className="spinner-border" role="status">
        <span className="sr-only">t('loader.label')</span>
      </div>
    </div>
  );
};
