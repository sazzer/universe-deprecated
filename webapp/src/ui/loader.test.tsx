import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import { Loader } from "./loader";

test("It renders correctly", () => {
  const { container } = render(<Loader />);
  expect(container).toMatchSnapshot();
});
