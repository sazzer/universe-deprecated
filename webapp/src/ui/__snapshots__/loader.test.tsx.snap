// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
<div>
  <div
    class="d-flex justify-content-center"
  >
    <div
      class="spinner-border"
      role="status"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
